proxy/cache.go

```diff
@@ -258,19 +258,22 @@ func filterMsg(dst, m *dns.Msg, ad, do bool, ttl uint32) {
 	dst.Extra = filterRRSlice(m.Extra, do, ttl)
 }
 
-// unpackResponse returns the unpacked response if it exists and didn't expire,
-// nil otherwise.
-func unpackResponse(data []byte, request *dns.Msg) *dns.Msg {
+// unpackResponse returns the unpacked response if it exists and if it is expired,
+func unpackResponse(data []byte, request *dns.Msg) (*dns.Msg, bool) {
 	expire := binary.BigEndian.Uint32(data[:4])
 	now := time.Now().Unix()
-	if int64(expire) <= now {
-		return nil
+
+	ttl := int64(expire) - now
+	isNotExpired := ttl > 0
+
+	// if expired,set the ttl to 60
+	if !isNotExpired {
+		ttl = 60
```

Comment:
This looks more like an optimistic cache and not cache prefetching.  Perhaps rename the configuration parameter as well?

Reply:
You are right, lol

Reply:
Is CacheOptimistic better?